File: payment-process-frontend/src/app/services/payment.service.ts
import { Payment } from './../model/payment.model';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { PATH_API } from './service';


@Injectable({
    providedIn: 'root'
})
export class PaymentService {

    constructor(private http: HttpClient) { }

    create(payment: Payment) {
        return this.http.post(`${PATH_API}/payment`, payment);
    }

    findAll() {
        return this.http.get(`${PATH_API}/payment`);
    }

    findById(id: string) {
        return this.http.get(`${PATH_API}/payment/${id}`);
    }
    
    findPaymentHistoryById(id: string) {
        return this.http.get(`${PATH_API}/payment/history/${id}`);
    }

}

File: payment-process-api/src/main/java/io/swagger/service/PaymentService.java
package io.swagger.service;

import java.util.List;

import org.springframework.stereotype.Component;

import io.swagger.api.response.Response;
import io.swagger.model.Payment;

@Component
public interface PaymentService {
	
	Payment createOrUpdate(Payment payment);

	Response<List<Payment>> findAll();
	
	void process(Payment payment);

	Response<Payment> create(Payment payment);

	Response<Payment> findById(Long id);
}
